tests/test_ssl_transport_close.py

```diff
@@ -0,0 +1,83 @@
+from aiohttp import web
+import aiohttp
+import asyncio
+import warnings
+
+
+async def handle_root(request):
+    """
+    idle 2 seconds so service is restarted meanwhile, forcing connection_lost
+    """
+    await asyncio.sleep(2)
+    return web.Response(text='', status=200)
+
+
+async def _prepare(ssl_ctx, client_ssl_ctx):
+    app = web.Application()
+    app.router.add_get('/', handle_root)
+    runner = web.AppRunner(app)
+    await runner.setup()
+    site = web.TCPSite(runner, 'localhost', 12345, ssl_context=ssl_ctx)
+    await site.start()
+    asyncio.ensure_future(_restart(runner))
+    session = aiohttp.ClientSession(connector=aiohttp.TCPConnector(ssl=client_ssl_ctx))
+    return app, runner, session
+
+
+async def _close(app, runner, session):
+    await runner.shutdown()
+    await runner.cleanup()
+    await session.close()
+
+been_restarted = 0
+
+async def _restart(runner):
+    """
+    restart service every second to force connection_lost
+    """
+    global been_restarted
+    while True:
+        try:
+            for s in list(runner.sites):
+                await s.stop()
+                await s.start()
+            been_restarted += 1
+        except Exception as e:
+            print(e)
+        finally:
+            await asyncio.sleep(0.5)
+
+def _ssl_resource_warnings(w):
+    if w.category == ResourceWarning and w.filename.endswith('sslproto.py') and \
+        w.message.args[0].startswith('unclosed transport <asyncio.sslproto._SSLProtocolTransport object at'):
+        return w
+    return None
```

Comment:
So the proper implementation would be
```suggestion
    unclosed_transport_msg = 'unclosed transport <asyncio.sslproto._SSLProtocolTransport object at'
    return (
        w.category == ResourceWarning
        and w.filename.endswith('sslproto.py')
        and w.message.args[0].startswith(unclosed_transport_msg)
    )
```